### DISCUSSION TITLE
(Termux) Android drag and drop solution

### DISCUSSION THREAD

**Discussion by tarkeez** [coder/code-server#6679] [Q&A]
code-server on Android 12 through Termux, on Firefox and Chrome.

Unable to drag views (reorder tabs, drag extension to panel etc).

Does anyone know an Android browser that supports drag and drop?

Or an alternative solution?

Have tried external mouse, virtual mouse apps.

Is there another way to move an extension view to the right side panel? Maybe some file setting?

Thanks.

**Comment by zongou** [comment]
this is what i did
https://github.com/microsoft/vscode/commit/f8acd5e5d4d151ee56424485ebab66f3c3a30e3b

**Reply by zongou** [reply]
i have built packges in my vscode fork release page

**Reply by tarkeez** [reply]
Thanks for your reply. Does this enable drag and drop in Android? I had the impression after searching that is was not possible.

Fwiw, for my use case (moving an extension's view to the right sidebar) I ended up typing "move view" in the command palette and was it worked.

**Reply by zongou** [reply]
dont know what you mean by draging extenions to panel, to install ? no i tried it never worked. but if you refer to that drag n drop support that worked on vscode 1.8.6(maybe older), it did reenable this

**Reply by tarkeez** [reply]
No, I had an extension for a code helper ai (cursor) and it defaulted to the main left sidebar, but I wanted it in the right/extra sidebar so I could have it open while using the file explorer in the left sidebar. On desktop you do this by dragging and dropping the "view" (as they call it) to where you want it, but on android I couldn't drag and drop. It wasn't really a problem related to vscode/code-server for me (the fact that I was running the server on the android device in termux is irrelevant, I realized), but the it was a client problem (android unable to drag and drop).

So it turned out they have a searchable option called "move view" where you can select a "view" and move it. Won't let you reorder tabs etc but at least I got the main problem solved.

Ultimately code-server on android was a pain for various reasons and I gave up the idea of using my tablet for coding and just use a laptop when I'm out lol.

**Reply by zongou** [reply]
Drag n drop was working with older versions default. 
![IMG_20240416_022441.jpg](https://github.com/coder/code-server/assets/23202896/a407448a-1ee2-4617-8a69-39900a797d99)

but drag n drop to ' move view' had not

**Reply by tarkeez** [reply]
Ah, I see. So you made drag and drop work again for android by reverting that commit, but drag and drop for views never did, and still doesn't
